all_test.go

```diff
@@ -1424,3 +1441,40 @@ func TestAuthArchiveIdentity(t *testing.T) {
 		t.Fatalf("TestArchiveIdentity failed, archived user should not be allowed to be rename identity: %v", err)
 	}
 }
+
+// Implemented in a smilar way to TestAuthUpdateDuplicateIdentity
+func TestAccountLocking(t *testing.T) {
+	c := setup(t)
+	defer Teardown(c)
+	c.MaxFailedLoginAttempts = 4
+	c.EnableAccountLocking = true
+	c.loginDelayFactor = 0 // add 0 ms per invalid login attempt
+	c.Auditor = c
+
+	user, userErr := c.GetUserFromIdentity(joeEmail)
+	if userErr != nil {
+		t.Fatal("Couldn't find setup user")
+	}
+
+	loginAttempts := 7
+	for i := 0; i < 2; i++ {
```

Comment:
This is the correct structure that I was referring to, but just some small changes here would make the code more readable:
1. Don't call it `i`. Instead, call it `lockPass`, as in multiple passes over the same concept.
2. In the outer loop, have two branches in the if statement:

```go
if lockPass ==0 {
    // Enable locking
} else {
    // Disable locking, and assume that the previous pass left the user in a locked state
}
```